core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/DataInputPort.scala

```diff
@@ -0,0 +1,35 @@
+package edu.uci.ics.amber.engine.architecture.messaginglayer
+
+import edu.uci.ics.amber.engine.architecture.messaginglayer.DataInputPort.InternalDataMessage
+import edu.uci.ics.amber.engine.common.ambermessage.neo.{DataEvent, InternalMessage}
+import edu.uci.ics.amber.engine.common.ambertag.neo.Identifier
+
+import scala.collection.mutable
+
+object DataInputPort {
+  final case class InternalDataMessage(
+      from: Identifier,
+      sequenceNumber: Long,
+      command: DataEvent
+  ) extends InternalMessage
+}
+
+class DataInputPort(tupleProducer: TupleProducer) {
+  private val dataOrderingEnforcer =
+    new mutable.AnyRefMap[Identifier, OrderingEnforcer[DataEvent]]()
```

Comment:
OrderingEnforcer shouldn't take DataEvent. It should take InternalDataMessage. We are anyways giving it both